diff --git a/themes/agnoster.zsh-theme b/themes/agnoster.zsh-theme
index 518a14a3..e0997052 100644
--- a/themes/agnoster.zsh-theme
+++ b/themes/agnoster.zsh-theme
@@ -201,11 +201,20 @@ prompt_dir() {
   prompt_segment blue $CURRENT_FG '%~'
 }
 
+prompt_tmux() {
+if [[ ! -z "${TMUX}" ]]
+then
+#	INFO=$(tmux display-message -p "#S #I #P")
+	INFO=""
+	prompt_segment white $CURRENT_FG $INFO
+fi
+}
+
 # Virtualenv: current working virtualenv
 prompt_virtualenv() {
   local virtualenv_path="$VIRTUAL_ENV"
   if [[ -n $virtualenv_path && -n $VIRTUAL_ENV_DISABLE_PROMPT ]]; then
-    prompt_segment blue black "(`basename $virtualenv_path`)"
+    prompt_segment cyan black "(`basename $virtualenv_path`)"
   fi
 }
 
@@ -216,7 +225,7 @@ prompt_virtualenv() {
 prompt_status() {
   local -a symbols
 
-  [[ $RETVAL -ne 0 ]] && symbols+="%{%F{red}%}✘"
+  [[ $RETVAL -ne 0 ]] && symbols+="%{%F{red}%}✘ $RETVAL"
   [[ $UID -eq 0 ]] && symbols+="%{%F{yellow}%}⚡"
   [[ $(jobs -l | wc -l) -gt 0 ]] && symbols+="%{%F{cyan}%}⚙"
 
@@ -239,6 +248,7 @@ prompt_aws() {
 ## Main prompt
 build_prompt() {
   RETVAL=$?
+	prompt_tmux
   prompt_status
   prompt_virtualenv
   prompt_aws
